src/_pytest/nodes.py

```diff
@@ -764,6 +764,11 @@ def reportinfo(self) -> Tuple[Union["os.PathLike[str]", str], Optional[int], str
 
     @cached_property
     def location(self) -> Tuple[str, Optional[int], str]:
+        """
+        Returns a tuple of ``(relfspath, lineno, testname)`` for this item
```

Comment:
im under the impression that we might want to finally add the named tuple for the locations, either as follow-up or here

Reply:
Yeah, then we could document the NamedTuple rather than repeat ourselves everywhere. Is that an OK change w.r.t compatibility? Should be all the same to the users.

Reply:
It may be a breaking change for certain implementors, so extra care is necessary, but I'm also under the impression that it is not too bad

Reply:
so it's OK by you to do within 7.x?

Reply:
If we do it within 7.x,then type annotations might have to carry a extra cost for the union depending on external implementation

Reply:
For the union?

Reply:
I'm not sure how we'd do that. For example, we couldn't type `Item.location` as a named tuple, since existing classes derived from `Item` could be returning a simple tuple.

We could guarantee that hooks get a named tuple (by initializing a named tuple before we pass the arg to the hook), but that's about it.

Did you have something else in mind?

Reply:
Right now I don't have any good ideas for it :(

Reply:
Let's then go ahead with this docs PR?

Reply:
@RonnyPfannschmidt would you be able to review and possibly merge this?

Reply:
thanks for the ping, creating a followup card